src/contracts/strategies/StrategyFactory.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import "./StrategyBase.sol";
+import "../permissions/Pausable.sol";
+
+/**
+ * @title Factory contract for deploying StrategyBase contracts for arbitrary ERC20 tokens
+ *        and automatically adding them to the StrategyWhitelist in EigenLayer.
+ * @author Layr Labs, Inc.
+ * @dev This may not be compatible with non-standard ERC20 tokens. Caution is warranted.
+ */
+contract StrategyFactory is OwnableUpgradeable, Pausable {
+
+    uint8 internal constant PAUSED_NEW_STRATEGIES = 0;
+
+    /// @notice EigenLayer's StrategyManager contract
+    IStrategyManager public immutable strategyManager;
+
+    StrategyBase public strategyImplementation;
+
+    ProxyAdmin public eigenLayerProxyAdmin;
+
+    // @notice Mapping token => strategy contract for the token
+    mapping(IERC20 => IStrategy) public tokenStrategies;
```

Comment:
Do we want to do a storage contract like we do for everything else?

Reply:
seems like a reasonable plan; I'll get that done shortly

Reply:
completed in https://github.com/Layr-Labs/eigenlayer-contracts/pull/522/commits/2fa5492c0be328630bd9815226c734e092ee8d5e which also created an interface file